core-bundle/src/Csp/CspParser.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Csp;
+
+use Nelmio\SecurityBundle\ContentSecurityPolicy\ContentSecurityPolicyParser;
+use Nelmio\SecurityBundle\ContentSecurityPolicy\DirectiveSet;
+use Nelmio\SecurityBundle\ContentSecurityPolicy\PolicyManager;
+
+class CspParser
+{
+    public function __construct(private readonly PolicyManager $policyManager)
+    {
+    }
+
+    public function parseHeader(string $header): DirectiveSet
+    {
+        $parser = new ContentSecurityPolicyParser();
+        $directiveSet = new DirectiveSet($this->policyManager);
+        $names = $directiveSet->getNames();
+
+        $directives = explode(';', $header);
```

Comment:
Can't we use the Symfony `HeaderUtils::split()` here?

Reply:
Not sure if that's necessary. As far as I understand the specs, only a `;` is a valid separator in a CSP header. All the other cases covered by `HeaderUtils::split()` are not relevant for CSP headers.